src/backend/pipeline/cont_analyze.c

```diff
@@ -2183,11 +2183,14 @@ proj_and_group_for_windows(SelectStmt *proc, SelectStmt *view, ContAnalyzeContex
 	}
 
 	/* Create and truncate projection and grouping on temporal expression */
+	context->hoisted_name = ARRIVAL_TIMESTAMP;
```

Comment:
What if the query looks something like `SELECT x::timestamp AS arrival_timestamp FROM stream WHERE arrival_timestamp > clock_timestamp() - INTERVAL '5 second'`?